typed-racket-lib/typed-racket/infer/infer-unit.rkt

```diff
@@ -547,6 +547,12 @@
                         [else empty])])
              (% cset-meet proc-c (cgen/flds context flds flds*)))]
 
+          ;; two prefab structs with the same key
+          [((Prefab: k flds) (Prefab: k* flds*))
+           #:when (equal? k k*)
+           ;; FIXME: should account for mutable fields
```

Comment:
We can't merge this without either a more-conservative approach or a correct approach here.